Assets/Scripts/NPCs/BaseAi.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+using Random = UnityEngine.Random;
+
+namespace NPCs
+{
+    [RequireComponent(typeof(NavMeshAgent))]
+    public abstract class BaseAi : MonoBehaviour
+    {
+        [Header("State Management")]
+        [SerializeField] public string currentState;
+        [SerializeField] public string defaultState;
+        
+        [HideInInspector] public NavMeshAgent agent;
+
+        public void Awake()
+        {
+            agent = Utils.GetRequiredComponent<NavMeshAgent>(this);
+            agent.avoidancePriority = Random.Range(0, 99);
```

Comment:
Basically some of the spawned AI will have randomly higher priority, so those with lower priority will avoid them (leads to better pathfinding around each other).

Reply:
If the goal is to make sure everyone has a unique number (as much as possible), maybe we can have a static int instead where we do `agent.avoidancePriority = staticint++ % 99;`? Would be fancy and more deterministic (rather than having some play sessions sometimes having AIs with the same prio.

Reply:
Great solution, just added it in!